backend/MetricsWriter.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Backend
+{
+    public class MetricsWriter
+    {
+        private static Guid _id = Guid.NewGuid();
+        private Timer _timer;
+
+        public MetricsWriter()
+        {
+        }
+
+        public Task StartAsync()
+        {
+            _timer = new Timer(GetMetrics,
+                               null,
+                               TimeSpan.FromSeconds(10),
+                               TimeSpan.FromSeconds(10));
+
+            return Task.CompletedTask;
+        }
+
+        public async Task StopAsync()
+        {
+            await _timer.DisposeAsync();
+        }
+
+        private static void GetMetrics(object _)
+        {
+            var memory = GetMemory();
+            var cpu = GetCpu();
+
+            var mem_used = ((memory.Used / memory.Total) * 100d);
+
+            var output = $"{{ 'id': {_id.ToString("D")}, 'cpu_used': {cpu.Used:F2}, 'mem_used': {mem_used:F2}, 'mem_total': {memory.Total:F2} }}";
```

Comment:
Does the silo object provide any identifying info we could reuse to correlate to the logs that orleans emits? Is there something that identifies the silo within a cluster?

Reply:
got rid of this in latest commit